pypesto/problem.py

```diff
@@ -89,15 +88,23 @@ def __init__(self,
         self.lb = np.array(lb).flatten()
         self.ub = np.array(ub).flatten()
 
+        self.lb_full = np.array(lb).flatten()
+        self.ub_full = np.array(ub).flatten()
+
         self.dim_full = dim_full if dim_full is not None else self.lb.size
 
         if x_fixed_indices is None:
             x_fixed_indices = []
         self.x_fixed_indices = [int(i) for i in x_fixed_indices]
 
+        # We want the fixed values to be a list, since we might need to add
+        # or remove values during profile computation
         if x_fixed_vals is None:
-            x_fixed_vals = np.array([])
-        self.x_fixed_vals = np.array(x_fixed_vals)
+            x_fixed_vals = []
+        if not isinstance(x_fixed_vals, list):
```

Comment:
I assume a numpy array would not be changed to a list here. maybe ``x_fixed_vals = list(x_fixed_vals)`` is the thing to do here?